http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -24,22 +24,55 @@ import org.broadinstitute.dsde.workbench.model._
 import spray.json.{RootJsonFormat, RootJsonReader, _}
 import ca.mrvisser.sealerate
 
+import scala.util.matching.Regex
+
+sealed trait ContainerRegistry extends Product with Serializable {
+  def regex: Regex
+}
+object ContainerRegistry {
+  final case object GCR extends ContainerRegistry {
+    def regex: Regex = "(.*gcr.io)\\/(.*)\\/(.*)(\\:?.*)$".r
+  }
+  final case object DockerHub extends ContainerRegistry {
+    def regex: Regex = ".*\\/.*$".r
```

Comment:
Isn't this going to match any `/`-separated URL? How are we guarding against malicious ones?

Reply:
it's not ideal. But I don't think we can put more restriction because something like `nitesh1989/terra-jupyter-bioconductor` should be legit

Reply:
hey @adrazhi , just wanna cc you on this

Reply:
Hm, can we do better than `.*`? Maybe there are characters that are allowed in docker urls, like `[a-zA-z0-9_-]` or something. Would be good to prevent things like:
```
myrepo/mydocker; mysql -c "DROP ALL TABLES"; sudo rm -rf /; 
```
:)